### PR TITLE
Add VIA support to Prime_M. Clean up all files

### PR DESCRIPTION
**Pull request by holtenc**
This PR adds support for VIA. In addition I went through and cleaned things up to be in line with requested changes on my recent merged PRs. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by holtenc** [comment]
This should be my last PR for a good lone while. Thanks for your patience guys.

File: keyboards/primekb/prime_m/rules.mk [hunk 1]
```diff
@@ -14,22 +14,22 @@ BOOTLOADER = atmel-dfu
 # Build Options
 #   change yes to no to disable
 #
-BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration
-MOUSEKEY_ENABLE = yes       # Mouse keys
-EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = no        # Console for debug
-COMMAND_ENABLE = no        # Commands for debug and configuration
+BOOTMAGIC_ENABLE = no		# Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/primekb/prime_m/rules.mk]: Could you please change all these tabs back to spaces?

File: keyboards/primekb/prime_m/rules.mk [hunk 2]
```diff
@@ -14,22 +14,22 @@ BOOTLOADER = atmel-dfu
...
-BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality on B7 by default
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
-MIDI_ENABLE = no            # MIDI support
-UNICODE_ENABLE = no         # Unicode
-BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
-AUDIO_ENABLE = no           # Audio output on port C6
-FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
-HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs
+NKRO_ENABLE = yes			# USB Nkey Rollover
+BACKLIGHT_ENABLE = yes		# Enable keyboard backlight functionality on B7 by default
```
Comment (fauxpark) [inline keyboards/primekb/prime_m/rules.mk]:
```suggestion
BACKLIGHT_ENABLE = yes		# Enable keyboard backlight functionality
```

**Comment by holtenc** [comment]
Done

File: keyboards/primekb/prime_m/rules.mk (hunk 1 above)
Comment (holtenc) [reply keyboards/primekb/prime_m/rules.mk]: Done

File: keyboards/primekb/prime_m/readme.md [hunk 3]
```diff
@@ -4,12 +4,13 @@
 
 The Prime_M is a premium input device that is designed to be flexible so that it can cover many needs. The case is milled from a solid block of aluminum and features an integrated switch plate, which is 4.75mm thick yet designed to allow MX style switches to properly snap in. The bottom cover is machine from solid brass and is pre-drilled/countersunk for use with optional feet to give the device a slight tilt.
 
-Keyboard Maintainer: [MxBlue](https://github.com/MxBlu)  
-Hardware Supported: Prime_M PCB, ATMega32u2  
-Hardware Availability: [GB Post](https://geekhack.org/index.php?topic=91821.0), [B-Stock Listing](https://www.primekb.com/products/prime_m)
+* Keyboard Maintainer: [MxBlue](https://github.com/MxBlu)
```
Comment (noroadsleft) [inline keyboards/primekb/prime_m/readme.md]: You should add yourself as a maintainer.

```suggestion
* Keyboard Maintainer: [holtenc](https://github.com/holtenc), [MxBlue](https://github.com/MxBlu)
```

File: keyboards/primekb/prime_m/readme.md [hunk 4]
```diff
@@ -4,12 +4,13 @@
 
 The Prime_M is a premium input device that is designed to be flexible so that it can cover many needs. The case is milled from a solid block of aluminum and features an integrated switch plate, which is 4.75mm thick yet designed to allow MX style switches to properly snap in. The bottom cover is machine from solid brass and is pre-drilled/countersunk for use with optional feet to give the device a slight tilt.
 
-Keyboard Maintainer: [MxBlue](https://github.com/MxBlu)  
-Hardware Supported: Prime_M PCB, ATMega32u2  
-Hardware Availability: [GB Post](https://geekhack.org/index.php?topic=91821.0), [B-Stock Listing](https://www.primekb.com/products/prime_m)
+* Keyboard Maintainer: [MxBlue](https://github.com/MxBlu)  
+* Hardware Supported: Prime_M PCB, ATMega32u2
```
Comment (noroadsleft) [inline keyboards/primekb/prime_m/readme.md]:
```suggestion
* Hardware Supported: Prime_M PCB, ATMega32u2
```

File: keyboards/primekb/prime_m/config.h [hunk 5]
```diff
@@ -20,8 +21,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
-#define PRODUCT_ID      0x4024
+#define VENDOR_ID       0x5052
+#define PRODUCT_ID      0x504D
```
Comment (noroadsleft) [inline keyboards/primekb/prime_m/config.h]: Just for reference.

```suggestion
#define VENDOR_ID       0x5052 // "PR"
#define PRODUCT_ID      0x504D // "PM"
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**